frontend/server/libs/dao/Coder_Of_The_Month.dao.php

```diff
@@ -71,26 +71,27 @@ public static function calculateCoderOfTheMonth($startTime, $endTime) {
               SELECT
                 user_id,
                 MAX(time) latest_time,
-                rank
+                selected_by
               FROM
                 Coder_Of_The_Month
-              WHERE
-                rank = 1
               GROUP BY
                 user_id,
-                rank
+                selected_by
             ) AS cm on i.user_id = cm.user_id
+          LEFT JOIN
+            (SELECT user_id, time FROM Coder_Of_The_Month WHERE time = ? GROUP BY user_id) AS com on i.user_id
           WHERE
-            cm.user_id IS NULL
-            OR DATE_ADD(cm.latest_time, INTERVAL 1 YEAR) < ?
+            (cm.user_id IS NULL
+            OR DATE_ADD(cm.latest_time, INTERVAL 1 YEAR) < ?)
+            AND com.user_id IS NULL
           GROUP BY
             up.identity_id
           ORDER BY
             score DESC
           LIMIT 100
         ";
 
-        $val = [$startTime, $endTime, $endTime];
+        $val = [$startTime, $endTime, $endTime, $endTime];
```

Comment:
no debería ser esto

```suggestion
        $val = [$startTime, $endTime, $startTime, $endTime];
```

? el penúltimo fue el que agregaste y como que me suena a que debería ser `$startTime`. pero ya no estoy seguro :S

Reply:
Analizando la lógica, esto se ocupaba cuando se almacenaban varios `Coders` para el mismo mes. Actualmente ya sólo se almacena el `Coder` ganador, así que ya no es necesario.